src/controller/robot/controller.go

```diff
@@ -80,9 +80,14 @@ func (d *controller) Create(ctx context.Context, r *Robot) (int64, error) {
 		return 0, err
 	}
 
-	if r.ExpiresAt == 0 {
-		tokenDuration := time.Duration(config.RobotTokenDuration()) * time.Minute
-		r.ExpiresAt = time.Now().UTC().Add(tokenDuration).Unix()
+	var expiresAt int64
+	if r.Duration == -1 {
+		expiresAt = -1
+	} else if r.Duration == 0 {
+		// system default robot duration
+		expiresAt = time.Now().UTC().Add(time.Duration(config.RobotTokenDuration()) * 60 * 24 * time.Minute).Unix()
+	} else {
+		expiresAt = time.Now().UTC().Add(time.Duration(r.Duration) * 60 * 24 * time.Minute).Unix()
```

Comment:
`expiresAt = time.Now().AddDate(0, 0, r.Duration).Unix()` ?

No need to convert the time to UTC timezone because the timestamp is timezone independent.

Reply:
updated.